### PR TITLE
Return and fetch hard-coded JSON data.

### PR DESCRIPTION
**Pull request by eshika**
Step 3 of Week 3 Walkthrough

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -32,6 +32,20 @@ function showFunFact() {
  */
 async function fetchMessage() {
   const response = await fetch('/data');
-  const message = await response.text();
-  document.getElementById('message-container').innerText = message;
+  const message = await response.json();
+  console.log(message);
```
Comment (tristan-jh) [inline portfolio/src/main/webapp/script.js:36]: left-over from debugging?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,9 +26,35 @@
...
-    response.getWriter().println("Thanks for visiting my website!");
+    String json = convertToJsonUsingGson(messages);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts a ArrayList of messages into a JSON string using the Gson library.
+   */
+  private String convertToJsonUsingGson(ArrayList messages) {
```
Comment (tristan-jh) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: Keep the type-parameter on ArrayList, so ArrayList<String> in this case.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -24,9 +26,35 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
```
Comment (bheff-work) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: It's best to use the interface 'List' when possible. Gives you more flexibility

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -32,6 +32,20 @@ function showFunFact() {
...
+  const messageContainer = document.getElementById('message-container')
+  messageContainer.innerText = '';
+  for (let i = 0; i < message.length; i++) {
+    messageContainer.appendChild(createListElement(message[i]));
+  }
+
```
Comment (bheff-work) [inline portfolio/src/main/webapp/script.js:42]: nit: remove extra line

**Pull request merged**